### PR TITLE
KAFKA-2788: Allow specifying principals with comman in ACL CLI.

### PR DESCRIPTION
**Pull request by Parth-Brahmbhatt**

### CODE REVIEW

**Comment by Parth-Brahmbhatt** [comment]
In addition to allowing , in principal option, I made some minor changes like not using a single instance of authorizer which resulted in list after add/remove reporting inconsistent acls sometimes. I also added all as a valid option for all resources.

File: core/src/main/scala/kafka/admin/AclCommand.scala [hunk 1]
```diff
@@ -284,23 +290,21 @@ object AclCommand {
       .defaultsTo(All.name)
       .withValuesSeparatedBy(Delimiter)
 
-    val allowPrincipalsOpt = parser.accepts("allow-principals", "Comma separated list of principals where principal is in principalType:name format." +
+    val allowPrincipalsOpt = parser.accepts("allow-principal", "principal is in principalType:name format." +
       " User:* is the wild card indicating all users.")
       .withRequiredArg
       .describedAs("allow-principals")
```
Comment (junrao) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: allow-principals should be allow-principal. Ditto for the description of "deny-principals".

File: core/src/main/scala/kafka/admin/AclCommand.scala [hunk 2]
```diff
@@ -284,23 +290,21 @@ object AclCommand {
...
       .ofType(classOf[String])
-      .withValuesSeparatedBy(Delimiter)
 
-    val denyPrincipalsOpt = parser.accepts("deny-principals", "Comma separated list of principals where principal is in " +
-      "principalType: name format. By default anyone not in --allow-principals list is denied access. " +
+    val denyPrincipalsOpt = parser.accepts("deny-principal", "principal is in principalType: name format. " +
+      "By default anyone not in --allow-principal list is denied access. " +
```
Comment (junrao) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: We need to remove "list".

**Comment by junrao** [comment]
Also, not sure that I understand the issue with making authorizer a single instance. To list after add, one will need to run the AclCommand again, which will create a new instance of authorizer, right?

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 1 above)
Comment (Parth-Brahmbhatt) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: done.

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 2 above)
Comment (Parth-Brahmbhatt) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: done.

**Comment by junrao** [comment]
Also, in AclCommandTest, could you simulate an ssl user by including comma in the user name?

**Comment by Parth-Brahmbhatt** [comment]
@junrao I caught my self following a pattern where I would add,remove a principal and then execute list to check what was the final out come so I added that as the default behavior i.e. Everytime we add,remove acls successfully we immediately execute list and show the output to user. 

So without the reinitialization I saw every now and then I run into following situation:

bin/kafka-acls.sh --authorizer kafka.security.auth.SimpleAclAuthorizer --authorizer-properties zookeeper.connect=localhost:2181 --add --allow-principal User:test1 --allow-principal User:test3 --allow-hosts host1 --deny-principal User:BadBob --deny-hosts bad-host --operations Read --topic test1 --deny-principal User:Alice
Adding following acls for resource: Topic:test1
    User:BadBob has Deny permission for operations: Read from hosts: bad-host
    User:test1 has Allow permission for operations: Read from hosts: host1
    User:test3 has Allow permission for operations: Read from hosts: host1
    User:Alice has Deny permission for operations: Read from hosts: bad-host

Following is list of acls for resource: Topic:test1

Which is a confusing behavior. I can remove the list after add/remove but that is just pushing the problem to user. Let me know if you think otherwise.

**Comment by Parth-Brahmbhatt** [comment]
Added SSL principal in AClCommandTest

**Comment by junrao** [comment]
Thanks for the patch. LGTM

**Pull request closed without merging**